packages/lite/src/jlite.ts

```diff
@@ -1,32 +1,76 @@
 import type { ServiceManager } from '@jupyterlab/services';
+import { PageConfig } from '@jupyterlab/coreutils';
 import { JupyterLiteServer } from '@jupyterlite/server';
 
-const serverExtensions = [
-  import('@jupyterlite/pyolite-kernel-extension'),
-  import('@jupyterlite/server-extension'),
-];
+const serverExtensions = [import('@jupyterlite/server-extension')];
 
-export async function startJupyterLiteServer(): Promise<ServiceManager> {
-  const litePluginsToRegister: JupyterLiteServer.IPluginModule[] = [];
+async function createModule(scope: string, module: string) {
+  try {
+    const factory = await (window as any)._JUPYTERLAB[scope].get(module);
+    return factory();
+  } catch (e) {
+    console.warn(`Failed to create module: package: ${scope}; module: ${module}`);
+    throw e;
+  }
+}
+
+/**
+ * Iterate over active plugins in an extension.
+ */
+function* activePlugins(extension: any) {
+  // Handle commonjs or es2015 modules
+  let exports;
+  if (Object.prototype.hasOwnProperty.call(extension, '__esModule')) {
+    exports = extension.default;
+  } else {
+    // CommonJS exports.
+    exports = extension;
+  }
+
+  const plugins = Array.isArray(exports) ? exports : [exports];
+  for (const plugin of plugins) {
+    yield plugin;
+  }
+}
 
+type LiteServerConfig = {
+  federatedExtensions: Record<string, any>;
+  litePluginSettings: Record<string, any>;
+};
+
+export async function startJupyterLiteServer(config?: LiteServerConfig): Promise<ServiceManager> {
   /**
-   * Iterate over active plugins in an extension.
+   * This is sufficent to  initialise the global object?
    */
-  function* activePlugins(extension: any) {
-    // Handle commonjs or es2015 modules
-    let exports;
-    if (Object.prototype.hasOwnProperty.call(extension, '__esModule')) {
-      exports = extension.default;
-    } else {
-      // CommonJS exports.
-      exports = extension;
-    }
+  PageConfig.getOption('');
 
-    const plugins = Array.isArray(exports) ? exports : [exports];
-    for (const plugin of plugins) {
-      yield plugin;
-    }
-  }
+  /**
+   * Do not rely on a configuration being on the document body, accept configuration via arguments
+   * and set options on the page config directly
+   */
+  PageConfig.setOption(
+    'litePluginSettings',
+    JSON.stringify({
+      '@jupyterlite/pyodide-kernel-extension:kernel': {
+        pipliteUrls: ['https://unpkg.com/@jupyterlite/pyodide-kernel@0.0.6/pypi/all.json'],
+      },
+    }),
+  );
```

Comment:
we should be able to set the `litePluginSettings` here? and this would be ahead of the `@jupyterlite/pyodide-kernel-extension` being activated, so it should be able to pick up on these?

Reply:
> so it should be able to pick up on these?

Yes I would assume so.

Reply:
cool - and yes I am seeing some impact of setting this but it's not loading the `piplite` wheel from `unpkg` for some reason.

i.e. **with** setting the `litePluginSettings` here, on initializing the kernel two of the wheels are loaded from unpkg.com 👏 **but** it still looks to the `/pypi/all.json` locally and still loads `piplite` from there 🤔

![image](https://user-images.githubusercontent.com/1473646/229927130-6e63b4fa-45bd-4331-aa12-af05c4375719.png)

Reply:
ok so this kind of makes sense as `litePluginSettings` is setting the `pipliteUrls` i.e. where piplite will load it's wheels from -- we need to also configure something separate to direct where the `piplite` wheel itself will be loaded from...?

Reply:
opened https://github.com/jupyterlite/pyodide-kernel/pull/42 to address this

Reply:
I have also started looking at the remote federated module loading, which I think I'll start in a separate PR on top of this one.

I'm just wondering though, if we put that remote loading in place it will immediately supercede this arrangement of build-time including the server extension while picking up the wheels from online - so it's just better to move to that now. Then AFAIK this will mean that the kernel can be specified by option at runtime, and also be interchangeable (i.e. we can load other xeus kernels too)